.github/workflows/lint.yml

```diff
@@ -1,18 +1,59 @@
-name: PR - Lint
+# MDM-API Git Hub Actions
+#####################################
+# This GHA is responsible for initiating linting checks
+
+name: Quality Assurance - Lint
+run-name: Executing lint QA on ${{ github.repository }} 🚀
+
 on:
   pull_request:
     branches: [ main ]
+    paths:
+    - 'src/**'
+    - 'test/**'
+
+env:
+  environment: dev
+  timezone: 'Europe/London'
+  node: '18'
 
 jobs:
-  build:
+
+# 1. Setup test infrastructure
+  setup:
+    name: Infrastructure setup 🔧
     runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-node@v3
-      with:
-        node-version: '18'
-        cache: 'npm'
-    - name: Install modules
-      run: npm ci
-    - name: Run ESLint (TypeScript)
-      run: npm run lint
\ No newline at end of file
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+  
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Lint
+  lint:
+    name: Scanning 🎨
```

Comment:
should the name be "Linting"?

Reply:
It only scans for lint error, no linting is performed at GHA level.

Reply:
yeah, so something like "Lint scanning" - otherwise it can be security scanning, or something different. 
this so that we avoid ambiguity

Reply:
There will be no ambiguity as the Job name will be displayed in conjuction with the step name, so it will appear like this.

![image](https://user-images.githubusercontent.com/32125108/220099441-f29ee3b9-697e-4c10-83bf-6798659bf2d0.png)